test/integration/test_bitcoin_com_market.py

```diff
@@ -0,0 +1,450 @@
+        order_cancelled_event: OrderCancelledEvent = order_cancelled_event
+        self.assertEqual(order_cancelled_event.order_id, client_order_id)
+
+    def test_cancel_all(self):
+        trading_pair = "ETHBTC"
+        bid_price: Decimal = self.market.get_price(trading_pair, True) * Decimal("0.5")
+        ask_price: Decimal = self.market.get_price(trading_pair, False) * 2
+        amount: Decimal = Decimal("0.0001")
+        quantized_amount: Decimal = self.market.quantize_order_amount(trading_pair, amount)
+
+        # Intentionally setting invalid price to prevent getting filled
+        quantize_bid_price: Decimal = self.market.quantize_order_price(trading_pair, bid_price * Decimal("0.7"))
+        quantize_ask_price: Decimal = self.market.quantize_order_price(trading_pair, ask_price * Decimal("1.5"))
+
+        self.market.buy(trading_pair, quantized_amount, OrderType.LIMIT, quantize_bid_price)
+        self.market.sell(trading_pair, quantized_amount, OrderType.LIMIT, quantize_ask_price)
+        self.run_parallel(asyncio.sleep(1))
+        [cancellation_results] = self.run_parallel(self.market.cancel_all(5))
+        for cr in cancellation_results:
+            self.assertEqual(cr.success, True)
+
+    @unittest.skipUnless(any("test_list_orders" in arg for arg in sys.argv), "List order test requires manual action.")
+    def test_list_orders(self):
+        self.assertGreater(self.market.get_balance("ETH"), Decimal("0.01"))
+        trading_pair = "ETHBTC"
+        amount: Decimal = Decimal("0.0001")
+        quantized_amount: Decimal = self.market.quantize_order_amount(trading_pair, amount)
+
+        current_bid_price: Decimal = self.market.get_price(trading_pair, True)
+        bid_price: Decimal = current_bid_price * Decimal("0.8")
+        quantize_bid_price: Decimal = self.market.quantize_order_price(trading_pair, bid_price)
+
+        self.market.buy(trading_pair, quantized_amount, OrderType.LIMIT, quantize_bid_price)
+        self.run_parallel(asyncio.sleep(1))
+        [order_details] = self.run_parallel(self.market.list_orders())
+        self.assertGreaterEqual(len(order_details), 1)
+
+        self.market_logger.clear()
+
+    def test_deposit_info(self):
+        [deposit_info] = self.run_parallel(
+            self.market.get_deposit_info("ETH")
+        )
+        deposit_info: DepositInfo = deposit_info
+        self.assertIsInstance(deposit_info, DepositInfo)
+        self.assertGreater(len(deposit_info.address), 0)
+
+    @unittest.skipUnless(any("test_withdraw" in arg for arg in sys.argv), "Withdraw test requires manual action.")
+    def test_withdraw(self):
+        wallet = conf.test_erc20_token_address
+        currency = "ETH"
+        withdraw_amount = Decimal('0.02')
+
+        # Ensure the market account has enough balance for withdraw testing.
+        self.assertGreaterEqual(self.market.get_balance("ETH"), Decimal(withdraw_amount))
```

Comment:
From my understanding, the withdraw function in BCE uses the balance from the wallet account and not the trading account. 
With that in mind, since the `get_balance` here only checks the balance on the trading account, this assertion here is not really accurate.
You might want to look into the [`/api/2/account/balance`](https://api.exchange.bitcoin.com/#account-balance) endpoint instead.

Reply:
I have addressed this issue in this [commit](https://github.com/CoinAlpha/hummingbot/pull/1153/commits/2dfd7fca26264c8da3e71bf157902550bec50940).

Thanks for the feedback!